volmdlr/wires.py

```diff
@@ -844,7 +844,18 @@ def __init__(self, primitives, reference_path: str = PATH_ROOT, name: str = ''):
         self.primitives = primitives
         self.reference_path = reference_path
         PhysicalObject.__init__(self, name=name)
+        self.index_next = 0
 
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.index_next >= len(self.primitives):
+            self.index_next = 0
+            raise StopIteration
+        else:
+            self.index_next += 1
+            return self.primitives[self.index_next-1]
```

Comment:
Because in the previous if it raises an error, you don't need to use an else statement here. You can delete the indentation of the block after it.